io/LasReader.cpp

```diff
@@ -178,17 +238,38 @@ QuickInfo LasReader::inspect()
     qi.m_metadata = m_metadata;
 
     done(table);
+    m_stream.reset();
 
     return qi;
 }
 
 
 void LasReader::createStream()
 {
-    if (!m_streamIf)
-        m_streamIf.reset(new LasStreamIf(m_filename));
+    if (m_count == 0)
```

Comment:
This if-block here means that doing `pdal info --metadata s3://some-large-las-file` will also do range requests - if this if-block is removed then pdal info / pdal pipeline will always fetch the entire file, and only calling `QuickInfo LasReader::inspect()` will do the range requests. All of the other changes to this file are just there for the range requests codepath to work in general, whether that's used only for calls to LasReader::inspect or used more broadly.